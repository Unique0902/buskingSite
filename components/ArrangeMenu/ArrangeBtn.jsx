import React from 'react';

export default function ArrangeBtn({ type, handleClick, children }) {
  const handleBtnClick = () => {
    handleClick(type);
  };
  return (
    <button
      className={
        'font-sans text-black text-lg text-left py-1 px-4 hover:bg-gray-200'
      }
      onClick={handleBtnClick}
    >
      {children}
    </button>
  );
}
