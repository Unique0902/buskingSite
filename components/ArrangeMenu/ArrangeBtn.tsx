import React, { ReactNode } from 'react';
type Props = {
  handleClick: () => void;
  children: ReactNode;
};
export default function ArrangeBtn({ handleClick, children }: Props) {
  const handleBtnClick = () => {
    handleClick();
  };
  return (
    <button
      className={' text-lg text-left py-1 px-4 hover:bg-gray-200'}
      onClick={handleBtnClick}
    >
      {children}
    </button>
  );
}
