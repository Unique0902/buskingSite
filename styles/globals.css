@tailwind base;
@tailwind components;
@tailwind utilities;
/* TODO: 글꼴 불러오는 로딩 줄이거나 없애기 */
* {
  box-sizing: border-box;
  font-family: 'Pretendard';
}

@font-face {
  font-family: 'Pretendard';
  font-weight: 45 920;
  font-style: normal;
  font-display: swap;
  src: url('../assets/fonts/PretendardVariable.woff2')
    format('woff2-variations');
}

li {
  list-style: none;
}
body {
  margin: 0;
  @apply text-black dark:text-white dark:bg-slate-600;
}
p,
h1,
h2 {
  margin: 0;
}
ul {
  padding: 0;
}
