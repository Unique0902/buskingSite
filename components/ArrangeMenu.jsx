import React from 'react';
import ArrangeBtn from './ArrangeBtn';
import PopupWrapper from './PopupWrapper';

const ArrangeMenu = ({
  setIsShowArrangeMenu,
  results,
  setResults,
  isBusking,
}) => {
  const arrangeResults = (type) => {
    switch (type) {
      case 'title':
        results.sort(function (a, b) {
          if (a.title.toLowerCase() > b.title.toLowerCase()) return 1;
          else if (a.title.toLowerCase() < b.title.toLowerCase()) return -1;
          else return 0;
        });
        break;
      case 'artist':
        results.sort(function (a, b) {
          if (a.artist.toLowerCase() > b.artist.toLowerCase()) return 1;
          else if (a.artist.toLowerCase() < b.artist.toLowerCase()) return -1;
          else return 0;
        });
        break;
      case 'time':
        results.sort(function (a, b) {
          return a.id - b.id;
        });
        break;
      case 'cnt':
        results.sort(function (a, b) {
          return b.cnt - a.cnt;
        });
        break;
    }
    setResults([...results]);
  };
  const handleClick = (type) => {
    arrangeResults(type);
    setIsShowArrangeMenu(false);
  };
  return (
    <PopupWrapper
      handleClickOther={() => {
        setIsShowArrangeMenu(false);
      }}
      isLeft={false}
    >
      <section className=' border-b border-gray-600 border-solid flex flex-col pt-2 pb-2'>
        <ArrangeBtn
          onClick={handleClick}
          type={'title'}
          text={'제목 문자순 정렬'}
        />
        <ArrangeBtn
          onClick={handleClick}
          type={'artist'}
          text={'가수 문자순 정렬'}
        />
        <ArrangeBtn onClick={handleClick} type={'time'} text={'시간순 정렬'} />
        {isBusking && (
          <ArrangeBtn
            onClick={handleClick}
            type={'cnt'}
            text={'신청자순 정렬'}
          />
        )}
      </section>
    </PopupWrapper>
  );
};

export default ArrangeMenu;
