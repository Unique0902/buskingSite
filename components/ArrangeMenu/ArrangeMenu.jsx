import React from 'react';
import ArrangeBtn from './ArrangeBtn';
import PopupWrapper from '../PopupWrapper';

const ArrangeMenu = ({
  setIsShowArrangeMenu,
  results,
  setResults,
  isBusking,
}) => {
  const arrangeTypes = {
    title: 'title',
    artist: 'artist',
    time: 'time',
    cnt: 'cnt',
  };
  const arrangeResults = (type) => {
    const copiedResults = [...results];
    switch (type) {
      case 'title':
        copiedResults.sort((a, b) => {
          if (a.title.toLowerCase() > b.title.toLowerCase()) return 1;
          else if (a.title.toLowerCase() < b.title.toLowerCase()) return -1;
          else return 0;
        });
        break;
      case 'artist':
        copiedResults.sort((a, b) => {
          if (a.artist.toLowerCase() > b.artist.toLowerCase()) return 1;
          else if (a.artist.toLowerCase() < b.artist.toLowerCase()) return -1;
          else return 0;
        });
        break;
      case 'time':
        copiedResults.sort((a, b) => {
          return a.id - b.id;
        });
        break;
      case 'cnt':
        copiedResults.sort((a, b) => {
          return b.cnt - a.cnt;
        });
        break;
    }
    setResults(copiedResults);
  };
  const handleClickArrangeBtn = (type) => {
    arrangeResults(type);
    setIsShowArrangeMenu(false);
  };
  const handleClickOtherInPopupWrapper = () => {
    setIsShowArrangeMenu(false);
  };
  return (
    <PopupWrapper
      handleClickOther={handleClickOtherInPopupWrapper}
      isLeft={false}
    >
      <section className='flex flex-col pt-2 pb-2 border-b border-gray-600 border-solid '>
        <ArrangeBtn
          handleClick={handleClickArrangeBtn}
          type={arrangeTypes.title}
        >
          제목 문자순 정렬
        </ArrangeBtn>
        <ArrangeBtn
          handleClick={handleClickArrangeBtn}
          type={arrangeTypes.artist}
        >
          가수 문자순 정렬
        </ArrangeBtn>
        <ArrangeBtn
          handleClick={handleClickArrangeBtn}
          type={arrangeTypes.time}
        >
          시간순 정렬
        </ArrangeBtn>
        {isBusking && (
          <ArrangeBtn
            handleClick={handleClickArrangeBtn}
            type={arrangeTypes.cnt}
          >
            신청자순 정렬
          </ArrangeBtn>
        )}
      </section>
    </PopupWrapper>
  );
};

export default ArrangeMenu;
