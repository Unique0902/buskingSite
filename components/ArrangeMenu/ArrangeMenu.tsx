import React from 'react';

import ArrangeBtn from './ArrangeBtn';
import PopupWrapper from '../PopUp/PopupWrapper';
import { ArrangeOption } from '../../store/data/ArrangeOptions';

//TODO: 나중에 정렬한 결과가 서버에 반영될수있게 그리고 정렬한 결과가 musicbar에서 잘되는지 근데 그건 서버에서 처리해야될듯
//TODO: 드래그해서 노래 순서 바꿀수있게 하기 아니면 위아래 버튼클릭

type Props<T> = {
  setIsShowArrangeMenu: React.Dispatch<React.SetStateAction<boolean>>;
  setResults: React.Dispatch<React.SetStateAction<T[]>>;
  arrangeOptionArr: ArrangeOption<T>[];
};

const ArrangeMenu = <T,>({
  setIsShowArrangeMenu,
  setResults,
  arrangeOptionArr,
}: Props<T>) => {
  const arrangeResults = (compareFunc: (a: T, b: T) => number) => {
    setResults((prevData) => [...prevData].sort(compareFunc));
  };

  return (
    <PopupWrapper
      handleClickOther={() => setIsShowArrangeMenu(false)}
      isLeft={false}
    >
      <section className='flex flex-col pt-2 pb-2 border-b border-gray-600 border-solid '>
        {arrangeOptionArr.map((data) => (
          <ArrangeBtn
            key={data.name}
            handleClick={() => {
              arrangeResults(data.arrangeFunc);
              setIsShowArrangeMenu(false);
            }}
          >
            {data.name}
          </ArrangeBtn>
        ))}
      </section>
    </PopupWrapper>
  );
};

export default ArrangeMenu;
