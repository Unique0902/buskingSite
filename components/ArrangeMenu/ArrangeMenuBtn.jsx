import React, { useState } from 'react';
import { borderRadius, color, xyPadding } from '../../styles/theme';
import ArrangeMenu from './ArrangeMenu';
import PrimaryBtn from '../Btn/PrimaryBtn';

const ArrangeMenuBtn = ({ results, setResults, isBusking }) => {
  const [isShowArrangeMenu, setIsShowArrangeMenu] = useState(false);
  const handleClick = () => {
    setIsShowArrangeMenu(true);
  };
  return (
    <div className='relative'>
      <PrimaryBtn
        handleClick={handleClick}
        bgColor={color.gray_400}
        radius={borderRadius.xm}
        btnPadding={xyPadding.base}
      >
        정렬
      </PrimaryBtn>
      {isShowArrangeMenu && (
        <ArrangeMenu
          setIsShowArrangeMenu={setIsShowArrangeMenu}
          results={results}
          setResults={setResults}
          isBusking={isBusking}
        />
      )}
    </div>
  );
};

export default ArrangeMenuBtn;
