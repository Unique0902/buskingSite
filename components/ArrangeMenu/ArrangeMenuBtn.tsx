import React, { useState } from 'react';
import { borderRadius, color, xyPadding } from '../../styles/theme';
import ArrangeMenu from './ArrangeMenu';
import PrimaryBtn from '../Btn/PrimaryBtn';
import { PlaylistSongData } from '../../store/type/playlist';
import { ApplianceData } from '../../store/type/busking';
type Props = {
  results: PlaylistSongData[] | ApplianceData[];
  setResults: React.Dispatch<
    React.SetStateAction<PlaylistSongData[] | ApplianceData[]>
  >;
  isBusking: boolean;
};
const ArrangeMenuBtn = ({ results, setResults, isBusking }: Props) => {
  const [isShowArrangeMenu, setIsShowArrangeMenu] = useState<boolean>(false);
  const handleClick = () => {
    setIsShowArrangeMenu(true);
  };
  return (
    <div className='relative'>
      <PrimaryBtn
        handleClick={handleClick}
        bgColor={color.gray_400}
        radius={borderRadius.xm}
        btnPadding={xyPadding.base}
      >
        정렬
      </PrimaryBtn>
      {isShowArrangeMenu && (
        <ArrangeMenu
          setIsShowArrangeMenu={setIsShowArrangeMenu}
          results={results}
          setResults={setResults}
          isBusking={isBusking}
        />
      )}
    </div>
  );
};

export default ArrangeMenuBtn;
