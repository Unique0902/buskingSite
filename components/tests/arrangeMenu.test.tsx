import { describe, expect, it, jest } from '@jest/globals';
import ArrangeMenu from '../ArrangeMenu/ArrangeMenu';
import renderer from 'react-test-renderer';

describe('arrangeMenu component test', () => {
  const setIsArrangeMenu = jest.fn();
  const setResults = jest.fn();
  const arrangeOption = [
    {
      name: '제목 문자순 정렬',
      arrangeFunc: (a: number, b: number) => a - b,
    },
  ];
  it('renders correctly', () => {
    const component = renderer.create(
      <ArrangeMenu
        setIsShowArrangeMenu={setIsArrangeMenu}
        setResults={setResults}
        arrangeOptionArr={arrangeOption}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
