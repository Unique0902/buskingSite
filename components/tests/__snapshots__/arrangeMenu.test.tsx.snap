// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrangeMenu component test renders correctly 1`] = `
<div
  className="w-80 border-gray-600 border bg-white dark:bg-slate-800 absolute rounded-xl z-50 top-0 right-0 "
>
  <section
    className="flex flex-col pt-2 pb-2 border-b border-gray-600 border-solid "
  >
    <button
      className=" text-lg text-left py-1 px-4 hover:bg-gray-200"
      onClick={[Function]}
    >
      제목 문자순 정렬
    </button>
  </section>
</div>
`;
